src/ordering/service.py

```diff
@@ -1,8 +1,53 @@
+import logging
 from uuid import UUID
 
+from injector import inject
+
+from application.settings import Settings
+from currency import BTC, ExchangeRateService
+
 from .commands import CreateBuyOrder
+from .db import Repository
+from .errors import BalanceLimitExceeded, OrderAlreadyExists
+from .events import BuyOrderCreated
 
+log = logging.getLogger(__name__)
 
+
+@inject
 class Service:
+    def __init__(
+            self,
+            settings: Settings,
+            repository: Repository,
+            exchange_rates: ExchangeRateService,
+    ) -> None:
+        self._bought_btc_limit = settings.ordered_btc_limit
+        self._repository = repository
+        self._get_order_id = repository.get_order_id
```

Comment:
Actually, why assign a method of collaborating an object to an attribute on self? It makes it harder to navigate through code as clicking would first move me in PyCharm here, then after amazing ceases I would have to click + CMD for the second time to get to actual implementation (or rather, an interface and then navigate for the 3rd time)

Reply:
To make it shorter, but you are right. Even for readability in application service it's more clear to see that you get it from repository